starter-code/scripts/models/article.js

```diff
@@ -70,35 +70,45 @@
       return article.body.split(' ').length;
     })
     // TODO: complete this reduce to get a grand total word count
-    .reduce(function() {
+    .reduce(function(a, b) {
+      var sum = a + b;
+      // console.log(sum);
+      return sum;
     });
   };
 
   /* Chain together a `map` and a `reduce` call to
           produce an array of *unique* author names. */
   Article.allAuthors = function() {
   //return       TODO: map our collection
-    //return    TODO: return just the author names
+  //return    TODO: return just the author names
+    return Article.allArticles.map(function(article){
+      return article.author;
+    }).filter(function(author, index, array){
+      return array.indexOf(author) === index;
+    });
```

Comment:
Great use of filter! I hope the lecture we had on reduce() was helpful. There are a couple ways you could use reduce here. You saw Ricks way. When I did the assignment, this is how I implemented it: 
```
.reduce(function(acc, cur){
      if (!acc.includes(cur)){
        acc.push(cur);
      }
      return acc;
    }, []);
```